### ISSUE TITLE
[BUG - Job API] 'plugins.disable' is not respected into the Job API

### ISSUE THREAD

**Issue by willianrampazzo** [avocado-framework/avocado#4071]
Code to test:

```
import os

from avocado import Test
from avocado.core.job import Job


class TestDisable(Test):

    def test_pass(self):
        pass

if __name__ == '__main__':
    reference = '%s:TestDisable.test_pass' % __file__
    config = {'run.references': [reference],
              'plugins.disable': ['result.xunit'],
              }

    with Job(config) as j:
        j.run()
```

Output:

```
[wrampazz@wrampazz avocado.dev]$ python3 /tmp/test.py 
JOB ID     : 3a897bf4d845cfc2dc49a16a4b1af09d1d318629
JOB LOG    : /home/wrampazz/avocado/job-results/job-2020-08-07T18.01-3a897bf/job.log
 (1/1) /tmp/test.py:TestDisable.test_pass: PASS (0.00 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0

[wrampazz@wrampazz avocado.dev]$ ls /home/wrampazz/avocado/job-results/job-2020-08-07T18.01-3a897bf/
id  jobdata  job.log  results.json  results.tap  results.xml  sysinfo  test-results
[wrampazz@wrampazz avocado.dev]$
```

Expected behavior: the `results.xml` file is not available into the results directory.

Running with a custom config file works as expected.

Config file:

```
[plugins]
disable = ['result.xunit']
```

Command-line:

```
[wrampazz@wrampazz avocado.dev]$ avocado --config=/tmp/avocado.conf run /bin/true
JOB ID     : 32ce1244668ce47198cf43d41c7974298ecdfa03
JOB LOG    : /home/wrampazz/avocado/job-results/job-2020-08-07T18.04-32ce124/job.log
 (1/1) /bin/true: PASS (0.01 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/wrampazz/avocado/job-results/job-2020-08-07T18.04-32ce124/results.html
JOB TIME   : 0.61 s

[wrampazz@wrampazz avocado.dev]$ ls /home/wrampazz/avocado/job-results/job-2020-08-07T18.04-32ce124/
id  jobdata  job.log  results.html  results.json  results.tap  sysinfo  test-results
[wrampazz@wrampazz avocado.dev]$
```

**Comment by beraldoleal** [comment]
This is also related to #4081. I'm working on this.

**Comment by clebergnu** [comment]
Fixed by #4084
